File: AI-Career-Coach-Backend/app/mock_interview_v2/schemas/interview.py
from typing import Dict, Any, List, Optional
from pydantic import BaseModel, Field
from datetime import datetime


class InterviewStartRequestV2(BaseModel):
    interviewer_id: str = "female_hr"
    target_role: str = "Senior Software Engineer"
    job_description: Optional[str] = None
    required_skills: List[str] = []
    preferred_skills: List[str] = []
    experience_level: str = "Mid-Senior"
    interview_type: str = "Comprehensive HR & Technical"
    difficulty: str = "Hard"
    candidate_name: Optional[str] = "Candidate"
    resume_context: Optional[str] = None


class InterviewTurnRequestV2(BaseModel):
    session_id: str
    phase: str
    question_number: int
    question_text: str
    candidate_answer: str
    elapsed_seconds: Optional[int] = 0


class InterviewQuestionV2(BaseModel):
    id: str
    number: int
    phase: str
    question: str
    category: str
    expected_competency: str
    context_reason: Optional[str] = None
    created_at: str


class InterviewSessionResponseV2(BaseModel):
    success: bool = True
    session_id: str
    interviewer_id: str
    target_role: str
    difficulty: str
    status: str
    current_phase: str
    current_question: Optional[InterviewQuestionV2] = None
    created_at: str

File: AI-Career-Coach-frontend/src/modules/mock-interview-v2/types/interview.ts
export interface InterviewQuestionV2 {
  id: string;
  number: number;
  phase: string;
  question: string;
  category: string;
  expected_competency: string;
  audio_base64?: string;
  video_url?: string;
  created_at: string;
}

export interface InterviewSessionV2 {
  session_id: string;
  interviewer_id: string;
  interviewer_name?: string;
  target_role: string;
  difficulty: string;
  current_phase: string;
  question_number: number;
  status: "IN_PROGRESS" | "COMPLETED" | "ERROR";
  current_question?: InterviewQuestionV2 | null;
  created_at: string;
  completed_at?: string | null;
}

export interface TranscriptItemV2 {
  id: string;
  sender: "interviewer" | "candidate";
  senderName: string;
  text: string;
  timestamp: string;
  phase: string;
}
